tt_metal/tt_hal/hal_api.hpp

```diff
@@ -0,0 +1,18 @@
+// SPDX-FileCopyrightText: © 2025 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <cstdint>
+
+namespace tt::tt_metal::tt_hal {
+
+/**
+ * @brief Uses the hardware abstraction layer to inform client of architecture specific L1 Size
+ *
+ * @return Size in bytes of the L1 SRAM buffer associated with the currently present architecture.
+ */
+uint32_t get_l1_size();
```

Comment:
why do we need this when device has `l1_size_per_core()`?

Reply:
```
uint32_t Device::l1_size_per_core() const {
    return tt::Cluster::instance().get_soc_desc(id_).worker_l1_size;
}
```

1. I didn't know this existed.
2. Are device objects accessible to ttnn? Should they be? (Rephrased, is this a public API exported to consumers?). I don't see it here: https://github.com/tenstorrent/tt-metal/blob/main/tt_metal/include/tt_metal/device.hpp
3. What is this? It doesn't interact with the Hal at all? Getting information from a file instead?
4. Even if we don't need this particular API, we will likely need others like this one to solve:
https://github.com/orgs/tenstorrent/projects/102/views/1?pane=issue&itemId=89632723&issue=tenstorrent%7Ctt-metal%7C15688

Reply:
I guess it is there, it just looks different:

```
/**
 * @brief Retrieves the available L1 size per worker core on the device.
 *
 * @param device The device to query.
 * @return L1 size per core in bytes.
 */
std::uint32_t GetL1SizePerCore(DeviceHandle device);
```

Reply:
The ttnn code I had to modify did not have a DeviceHandle or alternatively a device object readily available for me to use that API.

Reply:
@ayerofieiev-tt I am not that familiar with `ttnn` code. Can you take a look at the changes I made to `ttnn/cpp/ttnn/operations/experimental/transformer/create_qkv_heads/device/create_qkv_heads_program_factory.cpp` and let me know if its possible for me to get access to a device handle at these program points?

Reply:
@ayerofieiev-tt @abhullar-tt I have an alternate approach in a parallel PR using the suggested API.

https://github.com/tenstorrent/tt-metal/pull/15806